src/app/components/services/qhana-plugin.service.ts

```diff
@@ -0,0 +1,171 @@
+    const head = document.querySelector('head');
+    data.urls.forEach((url) => {
+      const styleLink = document.createElement('link');
+      styleLink.href = url;
+      styleLink.rel = 'stylesheet';
+      head.appendChild(styleLink);
+    });
+    document.body.style.background = 'transparent';
+  }
+  
+  getUUID(impl: ImplementationItem, uuids: Map<string, string>): string {
+    const implID = `${impl.name} ver: ${impl.version} (${impl.download} ${impl.type})`
+
+    const res = uuids.get(implID);
+    if (res != null) {
+      return res;
+    }
+    
+    const newUUID = uuidv4();
+    uuids.set(implID, newUUID);
+    return newUUID;
+  }
+  
+  getAlgoUUID(impl: ImplementationItem): string {
+    return this.getUUID(impl, this.algoUUIDs);
+  }
+
+  getImplUUID(impl: ImplementationItem): string {
+    return this.getUUID(impl, this.implUUIDs);
+  }
+
+  /**
+   * Handle implementation messages that send links to quantum circuit implementations
+   *
+   * @param {{type: 'implementations-response', implementations: ImplementationItem[]}} data
+   */
+  onImplementationsResponse(data: {
+    implementations: ImplementationItem[];
+  }): void {
+    const implementationsDto = data.implementations.map((impl) => {
+      const algoId = this.getAlgoUUID(impl);
+      const implId = this.getImplUUID(impl);
+
+      let language: string
+      if(impl.type === 'qasm'){
+        language = 'OpenQASM';
+      } else {
+        language = 'Qiskit';
+      }
+      
+      return {
+        id: algoId,
+        algorithmName: `${impl.name} (v${impl.version})`,
+        implementedAlgorithm: implId,
+        name: `${impl.name} (v${impl.version})`,
+        language,
+        sdk: 'Qiskit',
+        fileLocation: impl.download,
+        selectionRule: '',
+      };
+    });
+
+    this.implementationDtoSubject.next(implementationsDto);
+  }
+
+  /**
+   * Send a message to the parent window.
+   *
+   * @param {string|object} message the data attribute of the created message event
+   */
+  sendMessage(message: string | object): void {
+    const targetWindow = window.opener || window.parent;
+    if (targetWindow) {
+      targetWindow.postMessage(message, '*');
+    } else {
+      console.warn(
+        'Failed to message parent window. Is this page loaded outside of an iframe?'
+      );
+    }
+  }
+
+  /**
+   * Measure the current height and inform the parent window if the height has changed.
+   */
+  notifyParentWindowOnHeightChange(): void {
+    if (this.isPlugin) {
+      const height = Math.max(
+        document.body.scrollHeight,
+        document.documentElement.scrollHeight
+      );
+      if (height !== this.qhanaFrontendState.height) {
+        this.sendMessage({ type: 'ui-resize', height });
+      }
+    }
+  }
+
+  initializePlugin(): void {
+    // prevent double execution if script is already loaded in the current window
+    if (!this.qhanaFrontendState.initialized) {
```

Comment:
This does not prevent double execution without the variable being changed somewhere!